R/lintr-deprecated.R

```diff
@@ -176,3 +176,21 @@ semicolon_terminator_linter <- function(semicolon = c("compound", "trailing")) {
   allow_trailing <- !"trailing" %in% semicolon
   semicolon_linter(allow_compound, allow_trailing)
 }
+
+#' Unnecessary concatenation linter
+#' @rdname lintr-deprecated
+#' @export
+unneeded_concatenation_linter <- function(allow_single_expression = TRUE) {
+  lintr_deprecated(
+    old = "unneeded_concatenation_linter",
+    new = "unnecessary_concatenation_linter",
+    version = "3.1.0",
+    type = "Linter"
+  )
+
+  stopifnot(
```

Comment:
isn't this check being duplicated?

Reply:
Happy to remove it. 

I deliberately duplicated it because when the user uses `unneeded_concatenation_linter()`, it can be confusing to see that the error message refers to `unnecessary_concatenation_linter()`: 

``` r
library(lintr)

unneeded_concatenation_linter(allow_single_expression = 1.0)
#> Warning: Linter unneeded_concatenation_linter was deprecated in lintr version
#> 3.1.0. Use unnecessary_concatenation_linter instead.
#> Error in unnecessary_concatenation_linter(allow_single_expression = allow_single_expression): is.logical(allow_single_expression) is not TRUE
```

<sup>Created on 2022-12-18 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
ok fair enough